### ISSUE TITLE
InfluxDB runs as root on RHEL/CentOS when installed from RPM

### ISSUE THREAD

**Issue by GaryRogers** [influxdb/influxdb#3083]
On RHEL/CentrOS hosts without `start-stop-daemon` influxdb runs as root rather than the user:group specified in the init script.

From `/scripts/init.sh`

``` shell
if which start-stop-daemon > /dev/null 2>&1; then
            start-stop-daemon --chuid $GROUP:$USER --start --quiet --pidfile $pidfile --exec $daemon -- -pidfile $pidfile -config $config $INFLUXD_OPTS >>$STDOUT 2>>$STDERR &
        else
            nohup $daemon -pidfile $pidfile -config $config $INFLUXD_OPTS >>$STDOUT 2>>$STDERR &
        fi
```

The else case isn't trying to do any daemon or su to change user on start. A quick hack is:

``` shell
su -c "nohup $daemon -pidfile $pidfile -config $config $INFLUXD_OPTS >>$STDOUT 2>>$STDERR &" influxdb
```

It would probably be better to source `/etc/rc.d/init.d/functions` and use the `daemon` function provided there. Not sure what impact that would have on the rest of the script though.

**Comment by biskit** [comment]
+1. something like 'su -l influxdb ...' or 'runuser -l' should do the trick. one needs to source /etc/rc.d/init.d/functions though.

pls refer to /etc/init.d/postgresql or some of the well known ones for reference.

also reference #3123

**Comment by otoolep** [comment]
This has been fixed in 0.9.3.
